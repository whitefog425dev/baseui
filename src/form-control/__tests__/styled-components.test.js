/*
Copyright (c) 2018 Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/

// @flow
import React from 'react';
import {shallow} from 'enzyme';
import {
  Label as StyledLabel,
  Caption as StyledCaption,
} from '../styled-components.js';

describe('FormControl - Label and Caption for controls', () => {
  test('StyledLabel - basic render', () => {
    const component = shallow(<StyledLabel>Label</StyledLabel>);
    expect(component.instance().getStyles()).toMatchSnapshot(
      'StyledLabel has correct default styles',
    );
    component.setProps({
      $disabled: true,
    });
    expect(component.instance().getStyles()).toMatchSnapshot(
      'StyledLabel has correct styles when compact and disabled',
    );
  });

  test('StyledCaption - basic render', () => {
    const component = shallow(<StyledCaption>Caption</StyledCaption>);
    expect(component.instance().getStyles()).toMatchSnapshot(
      'StyledCaption has correct default styles',
    );
    component.setProps({
      $error: true,
    });
    expect(component.instance().getStyles()).toMatchSnapshot(
      'StyledCaption has correct styles when compact and error is boolean',
    );
    component.setProps({
      $error: 'Error message',
    });
    expect(component.instance().getStyles()).toMatchSnapshot(
      'StyledCaption has correct styles when error is a string',
    );
  });
});
