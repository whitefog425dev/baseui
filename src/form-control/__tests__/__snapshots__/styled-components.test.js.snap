// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormControl - Label and Caption for controls StyledCaption - basic render: StyledCaption has correct default styles 1`] = `
Object {
  "color": "$theme.colors.foregroundAlt",
  "fontFamily": "$theme.typography.font200.fontFamily",
  "fontSize": "$theme.typography.font200.fontSize",
  "fontWeight": "$theme.typography.font200.fontWeight",
  "lineHeight": "$theme.typography.font200.lineHeight",
  "marginBottom": "$theme.sizing.scale300",
  "marginLeft": "0",
  "marginRight": "0",
  "marginTop": "$theme.sizing.scale300",
  "paddingBottom": "0",
  "paddingLeft": "0",
  "paddingRight": "0",
  "paddingTop": "0",
}
`;

exports[`FormControl - Label and Caption for controls StyledCaption - basic render: StyledCaption has correct styles when compact and error is boolean 1`] = `
Object {
  "color": "$theme.colors.foregroundAlt",
  "fontFamily": "$theme.typography.font200.fontFamily",
  "fontSize": "$theme.typography.font200.fontSize",
  "fontWeight": "$theme.typography.font200.fontWeight",
  "lineHeight": "$theme.typography.font200.lineHeight",
  "marginBottom": "$theme.sizing.scale300",
  "marginLeft": "0",
  "marginRight": "0",
  "marginTop": "$theme.sizing.scale300",
  "paddingBottom": "0",
  "paddingLeft": "0",
  "paddingRight": "0",
  "paddingTop": "0",
}
`;

exports[`FormControl - Label and Caption for controls StyledCaption - basic render: StyledCaption has correct styles when error is a string 1`] = `
Object {
  "color": "$theme.colors.negative400",
  "fontFamily": "$theme.typography.font200.fontFamily",
  "fontSize": "$theme.typography.font200.fontSize",
  "fontWeight": "$theme.typography.font200.fontWeight",
  "lineHeight": "$theme.typography.font200.lineHeight",
  "marginBottom": "$theme.sizing.scale300",
  "marginLeft": "0",
  "marginRight": "0",
  "marginTop": "$theme.sizing.scale300",
  "paddingBottom": "0",
  "paddingLeft": "0",
  "paddingRight": "0",
  "paddingTop": "0",
}
`;

exports[`FormControl - Label and Caption for controls StyledLabel - basic render: StyledLabel has correct default styles 1`] = `
Object {
  "color": "$theme.colors.foreground",
  "cursor": "auto",
  "display": "block",
  "fontFamily": "$theme.typography.font350.fontFamily",
  "fontSize": "$theme.typography.font350.fontSize",
  "fontWeight": 500,
  "lineHeight": "$theme.typography.font350.lineHeight",
  "marginBottom": "$theme.sizing.scale300",
  "marginLeft": "0",
  "marginRight": "0",
  "marginTop": "$theme.sizing.scale300",
  "paddingBottom": "0",
  "paddingLeft": "0",
  "paddingRight": "0",
  "paddingTop": "0",
}
`;

exports[`FormControl - Label and Caption for controls StyledLabel - basic render: StyledLabel has correct styles when compact and disabled 1`] = `
Object {
  "color": "$theme.colors.foregroundAlt",
  "cursor": "not-allowed",
  "display": "block",
  "fontFamily": "$theme.typography.font350.fontFamily",
  "fontSize": "$theme.typography.font350.fontSize",
  "fontWeight": 500,
  "lineHeight": "$theme.typography.font350.lineHeight",
  "marginBottom": "$theme.sizing.scale300",
  "marginLeft": "0",
  "marginRight": "0",
  "marginTop": "$theme.sizing.scale300",
  "paddingBottom": "0",
  "paddingLeft": "0",
  "paddingRight": "0",
  "paddingTop": "0",
}
`;
